### PR TITLE
Load reads by chunks of limited total size

### PR DESCRIPTION
**Pull request by salikhov-kamil**
@karel-brinda please, look at this request
do you think 10M size for chunk is OK?
maybe, it's better to add an option for it?

### CODE REVIEW

**Comment by karel-brinda** [comment]
Hi Kamil,

thank you for the PR. However, it seems that it does not compile (see Travis logs), could you look at the problems?

> maybe, it's better to add an option for it?

Yes, this is a great idea. Some sequencing experiments will definitely require special choices of the buffer size.

File: prophyle/prophyle_index/prophyle_query.c [hunk 1]
```diff
@@ -8,11 +8,17 @@
 #include "kstring.h"
 #include "klcp.h"
 #include "bwa_utils.h"
+#include "kseq.h"
 #include "contig_node_translator.h"
+KSEQ_DECLARE(gzFile)
 
 #define MAX_POSSIBLE_SA_POSITIONS 1000000
```
Comment (karel-brinda) [inline prophyle/prophyle_index/prophyle_query.c]: @salikhov-kamil Could you add comments to all these constants, please?

**Comment by salikhov-kamil** [comment]
@karel-brinda I added -r option for max read chunk size
do you think we need a test for it?
if yes, any suggestions how to test it?

**Comment by karel-brinda** [comment]
Kamil, you know how it works so you should think about all possible things that might go wrong and test them.

Btw. Could you put the default value into brackets? What's the unit? bp?

**Pull request merged**